src/MiniCover/Commands/BaseCommandLineApplication.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.Extensions.CommandLineUtils;
+using MiniCover.Commands.Options;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace MiniCover.Commands
+{
+    internal abstract class BaseCommandLineApplication : CommandLineApplication
+    {
+        private const string CommandHelpOption = "-h | --help";
+
+        protected BaseCommandLineApplication(CommandLineApplication parentCommandLineApplication)
+        {
+            Initialize();
+            Parent = parentCommandLineApplication;
+            HelpOption(CommandHelpOption);
+            InitializeOptions();
+            InitializeExecution();
+        }
+
+        protected abstract string CommandDescription { get; }
+        protected abstract string CommandName { get; }
+        protected abstract IEnumerable<IMiniCoverOption> MiniCoverOptions { get; }
+
+        protected abstract Task<int> Execution();
+
+        private void Initialize()
+        {
+            Name = CommandName;
+            Description = CommandDescription;
+        }
+
+        private void InitializeOptions()
+        {
+            foreach (var miniCoverOption in MiniCoverOptions)
+            {
+                if (miniCoverOption == null)
+                {
+                    throw new ArgumentNullException(nameof(miniCoverOption), "MiniCover option is null");
+                }
+                miniCoverOption.Initialize(this);
+            }
+        }
+
+        private void InvalidateOptions()
```

Comment:
ValidateOptions to match the other change

Reply:
@lucaslorentz done